frame/ethereum/src/lib.rs

```diff
@@ -632,10 +675,19 @@ impl<T: Config> Pallet<T> {
 		});
 
 		Ok(PostDispatchInfo {
-			actual_weight: Some(T::GasWeightMapping::gas_to_weight(
-				used_gas.unique_saturated_into(),
-				true,
-			)),
+			actual_weight: {
+				let mut gas_to_weight = T::GasWeightMapping::gas_to_weight(
+					used_gas.unique_saturated_into(),
+					true,
+					None,
+				);
+				if let Some(weight_info) = weight_info {
+					if let Some(proof_size_usage) = weight_info.proof_size_usage {
+						*gas_to_weight.proof_size_mut() = proof_size_usage;
```

Comment:
The returned actual weight should account for the transaction length in addition to the "used proof size", two possible solutions:

1. Add the encoded tx len here
2. Account the tx encoded lenght somewhere else ferore

Reply:
Yeah you are right, probably we want to account it first thing - as the equivalent of the evm transaction cost for pov size.